user-creation/src/main/java/no/unit/nva/useraccessservice/usercreation/person/IdentityServiceErrorCodes.java

```diff
@@ -0,0 +1,156 @@
+     * timeouts, connection refused, or other infrastructure-level failures.
+     */
+    public static final String SERVICE_UNAVAILABLE = "1002";
+
+    /**
+     * 1003: Missing National Identity Number (NIN).
+     * Used when a required National Identity Number is missing from user attributes.
+     */
+    public static final String MISSING_NIN = "1003";
+
+    // 2xxx - Resource-related errors (already exists, not found, etc.)
+
+    /**
+     * 2001: Person not found.
+     * Used when a requested person cannot be found in the registry.
+     */
+    public static final String PERSON_NOT_FOUND = "2001";
+
+    /**
+     * 2002: Person already exists.
+     * Used when attempting to create a person that already exists in the registry.
+     */
+    public static final String PERSON_ALREADY_EXISTS = "2002";
+
+    // 3xxx - Upstream service errors (integration or API failures)
+
+    /**
+     * 3001: Upstream internal server error.
+     * Used when the upstream person registry service responds with a 500+ error status.
+     */
+    public static final String UPSTREAM_INTERNAL_ERROR = "3001";
+
+    /**
+     * 3002: Upstream response parsing error.
+     * Used when the response from the upstream person registry service cannot be parsed
+     * (e.g., invalid JSON format or unexpected structure).
+     */
+    public static final String UPSTREAM_PARSING_ERROR = "3002";
+
+    /**
+     * 3003: Person creation failed.
+     * Used when person creation fails in the upstream registry for reasons other than
+     * the person already existing.
+     */
+    public static final String PERSON_CREATION_FAILED = "3003";
+
+    private IdentityServiceErrorCodes() {
+        // Utility class
+    }
+
+    /**
+     * Formats an error message as an RFC 7807 Problem Details JSON string.
+     *
+     * @param errorCode The error code constant (without IdentityService- prefix)
+     * @param message The error message
+     * @return JSON string representation of the Problem
+     */
+    public static String formatMessage(String errorCode, String message) {
+        try {
+            var problem = createProblem(errorCode, message, getStatusForErrorCode(errorCode));
+            return JsonUtils.dtoObjectMapper.writeValueAsString(problem);
+        } catch (JsonProcessingException e) {
+            // Fallback to a simple string format if JSON serialization fails
+            return ERROR_PREFIX + errorCode + ": " + message;
+        }
+    }
+
+    private static Problem createProblem(String errorCode, String message, Status status) {
+        return Problem.builder()
+            .withType(ERROR_TYPE_BASE_URI.addChild(errorCode).getUri())
+            .withTitle(ERROR_PREFIX + errorCode + ": " + message)
+            .withStatus(status)
+            .withDetail(message)
+            .build();
+    }
+
+    private static Status getStatusForErrorCode(String errorCode) {
```

Comment:
You can eliminate the need for this code with the ServiceErrorCode mentioned above. Either way, you're hard-coding the logic, and I think simpler mapping (no processor cycles on retrieval) is better and more readable.